src/server.rs

```diff
@@ -972,6 +972,22 @@ impl<B: IntoBuf> SendResponse<B> {
         self.inner.send_reset(reason)
     }
 
+    /// Polls to be notified when the client resets this stream.
+    ///
+    /// If stream is still open, this returns `Ok(Async::NotReady)`, and
```

Comment:
Maybe add a section above this: "This function allows the caller to detect a RST_STREAM before sending the response, providing an opportunity to cancel work. This function must only be called before calling `send_response`." or something like that. Basically, emphasize the intent / requirements.

Reply:
Both of those ideas are in the following lines. Are they not clear?

Reply:
It’s covered by the errors section but who reads that?